lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -83,14 +84,17 @@ def load_from_data(self, model, course_structure, scores_client, submissions_sco
         """
         for block in model.visible_blocks.blocks:
             persisted_values = {'weight': block.weight, 'possible': block.max_score}
+            locator = BlockUsageLocator.from_string(block.locator)
             self._compute_block_score(
                 User.objects.get(id=model.user_id),
-                block.locator,
+                locator,
```

Comment:
Using a string instead of a `BlockUsageLocator` here was causing `block = course_structure[block_key]` to return `None`, so `_compute_block_score` did nothing, and `self.scores` was loading as empty.

This is the odd behavior we were discussing this morning @nasthagiri